File: backend/app/models/project.py
from pydantic import BaseModel
from typing import List
from .note import NoteData

class ProjectMetadata(BaseModel):
    title: str
    instrument: str
    tempo: int
    time_signature: str = "4/4"
    key: str = "C"

class Project(BaseModel):
    version: str = "1.0"
    metadata: ProjectMetadata
    notes: List[NoteData]

File: frontend/src/types/project.ts
import { NoteData } from './note';

export interface ProjectMetadata {
  title: string;
  instrument: string;
  tempo: number;
  timeSignature: string;
  key: string;
  /** true when the meter came from U31 auto-detection (chip shows «(авто)») */
  timeSignatureAuto?: boolean;
}

export interface Project {
  version: string;
  metadata: ProjectMetadata;
  notes: NoteData[];
}

export interface TranscriptionData {
  notes: NoteData[];
  tempo: number;
  key: string;
  timeSignature: string;
  instrument: string;
  /** non-null when the backend auto-detected the meter (U31) */
  timeSignatureConfidence?: number | null;
}